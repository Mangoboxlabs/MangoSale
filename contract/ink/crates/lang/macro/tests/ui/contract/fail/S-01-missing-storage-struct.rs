use ink_lang as ink;

#[ink::contract]
mod missing_storage_struct {
    // We are missing the #[ink(storage)] attribute here
    pub struct MissingStorageStruct {}

    impl MissingStorageStruct {
        #[ink(constructor)]
        pub fn constructor() -> Self {}

        #[ink(message)]
        pub fn message(&self) {}
    }
}

fn main() {}
