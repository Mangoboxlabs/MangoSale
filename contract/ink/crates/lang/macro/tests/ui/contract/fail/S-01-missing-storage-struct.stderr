error: missing ink! storage struct
  --> $DIR/S-01-missing-storage-struct.rs:4:1
   |
4  | / mod missing_storage_struct {
5  | |     // We are missing the #[ink(storage)] attribute here
6  | |     pub struct MissingStorageStruct {}
7  | |
...  |
14 | |     }
15 | | }
   | |_^
